### ISSUE TITLE
R4.2: debian-11 fails to boot with in-vm kernel

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#8277]
## Observation

openQA test in scenario qubesos-4.2-update-x86_64-system_tests_pvgrub_salt_storage@64bit fails in
[TC_41_HVMGrub_debian-11](https://openqa.qubes-os.org/tests/75365/modules/TC_41_HVMGrub_debian-11/steps/1)

All three modes (PV, PVH, HVM) fail to boot with a timeout:
- PV: panic `Kernel panic - not syncing: System is deadlocked on memory`
- HVM: empty log
- PVH: some kernel errors are visible, but looks unrelated to boot failure

If fails with both StandaloneVM, and AppVM.

## Test suite description

Try booting debian-11 based VM with an in-vm kernel.

## Reproducible

Fails since (at least) Build [2023060606-4.2](https://openqa.qubes-os.org/tests/75192)

## Expected result

Last good: [2023053020-4.2](https://openqa.qubes-os.org/tests/74688) (or more recent)

## Further details

Always latest result in this scenario: [latest](https://openqa.qubes-os.org/tests/latest?arch=x86_64&distri=qubesos&flavor=update&machine=64bit&test=system_tests_pvgrub_salt_storage&version=4.2)

**Comment by marmarek** [comment]
This seems to be caused by initramfs having ~70MB compressed (~230MB uncompressed). Default initial 400MB of RAM is not enough to handle that.

128MB is used for modules (it isn't all of them, the whole modules dir in rootfs takes over 300MB), and then 57MB is used for firmware. Both could be much more limited, but I'm not sure how to do that in an elegant way (to not list all the modules verbatim as that could change between kernel versions and also to leave an option to override by the user).

**Comment by marmarek** [comment]
The same issue applies to debian-12 too.

**Comment by brendanhoar** [comment]
> This seems to be caused by initramfs having ~70MB compressed (~230MB uncompressed). Default initial 400MB of RAM is not enough to handle that.
> 

Is the issue directly the amount of RAM used by the uncompressed initramfs, or is it that *plus* the amount of *additional* RAM used during the decompression process?

I seem to recall reading the latter being an issue for raspberry pi units (some < 1GB RAM) in some distro's recent move to different/newer kernel compression algs, particularly at the higher compression settings...which turned out to need an additional 200MB+ in memory dictionary space at those high compression levels.

B

**Comment by marmarek** [comment]
Hmm, I just found out technically this is a duplicate of https://github.com/QubesOS/qubes-issues/issues/8235, and I guess "not our bug" resolution applies here too.

But at the some time, since I've identified it's about initramfs, actually there may be a solution: [initramfs.conf](https://manpages.debian.org/bullseye/initramfs-tools-core/initramfs.conf.5.en.html) has `MODULES=dep`  option that should reduce number of included modules (the default is `MODULES=most`). I'll try this approach.
